### PR TITLE
RHINENG-9370: migrate to app-common-js v2.2.4 

### PR DESCRIPTION
**Pull request by rexwhite**
Migrated to app-common-js@2.2.4 (breaking change) and added a basic test.

### CODE REVIEW

File: src/config/config.unit.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+const path = require('path');
+
+describe('Configuration', () => {
+    test('Verify app-common-js function', () => {
+        process.env.ACG_CONFIG = path.resolve('src/config/test.json');
+        const config = require('.');
+
+        expect(config).toMatchSnapshot();
+    });
+});
```
Comment (vkrizan) [inline src/config/config.unit.js:12]: Lack of newline at the EOF.

File: src/config/index.js [hunk 2]
```diff
@@ -47,9 +47,11 @@ function getHostForApp(dependencyEndpoints, appName, deploymentName, envar) {
 }
 
 function Config() {
-    const loadedConfig = (acgConfig) ? require('app-common-js').LoadedConfig : '';
-    const dependencyEndpoints = (acgConfig) ? require('app-common-js').DependencyEndpoints : '';
-    const privateDepencencyEndpoints = (acgConfig) ? require('app-common-js').PrivateDependencyEndpoints : '';
+    const app_config = (acgConfig) ? new (require('app-common-js').Config)() : null;
```
Comment (vkrizan) [inline src/config/index.js:50]: Prefer camelCase over snake_case in JS.

File: src/config/index.js [hunk 3]
```diff
@@ -103,14 +105,14 @@ function Config() {
 
         cache: {
             ttl: parseIntEnv('CACHE_TTL', 24 * 60 * 60), // 24 hours
-            revalidationInterval: parseIntEnv('CACHE_REVALIDATION_INVERVAL', 10 * 60) // 10 mins
+            revalidationInterval: parseIntEnv('CACHE_REVALIDATION_INTERVAL', 10 * 60) // 10 mins
```
Comment (vkrizan) [inline src/config/index.js:108]: Unrelated change. Please separate these in future, either by commit or PR. :pray:

**Review by vkrizan — APPROVED** [review]
Couple of files are lacking newline at the EOF, plus this bundles unrelated changes. 

Thanks.

File: src/config/index.js (hunk 3 above)
Comment (rexwhite) [reply src/config/index.js:108]: Yeah, I just stumbled across those, was already about 3 interruptions deep and didn't have the bandwidth to open another jira.  Your point is valid, though...

**Pull request merged**